Essentials/src/com/earth2me/essentials/services/ReserveEssentials.java

```diff
@@ -0,0 +1,1875 @@
+     * @param world The name of the {@link World} associated with the amount.
+     * @return True if the funds were set for the account, otherwise false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncSetHoldings(UUID identifier, BigDecimal amount, String world) {
+        return CompletableFuture.supplyAsync(() -> setHoldings(identifier, amount));
+    }
+
+    /**
+     * Used to set funds to an account.
+     * @param identifier The identifier of the account that is associated with this call.
+     * @param amount The amount you wish to set from this account.
+     * @param world The name of the {@link World} associated with the amount.
+     * @param currency The {@link Currency} associated with the balance.
+     * @return True if the funds were set for the account, otherwise false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncSetHoldings(String identifier, BigDecimal amount, String world, String currency) {
+        return CompletableFuture.supplyAsync(() -> setHoldings(identifier, amount));
+    }
+
+    /**
+     * Used to set funds to an account.
+     * @param identifier The identifier of the account that is associated with this call.
+     * @param amount The amount you wish to set from this account.
+     * @param world The name of the {@link World} associated with the amount.
+     * @param currency The {@link Currency} associated with the balance.
+     * @return True if the funds were set for the account, otherwise false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncSetHoldings(UUID identifier, BigDecimal amount, String world, String currency) {
+        return CompletableFuture.supplyAsync(() -> setHoldings(identifier, amount));
+    }
+
+    /**
+     * Used to add funds to an account.
+     * @param identifier The identifier of the account that is associated with this call.
+     * @param amount The amount you wish to add from this account.
+     * @return True if the funds were added to the account, otherwise false.
+     */
+    @Override
+    public boolean addHoldings(String identifier, BigDecimal amount) {
+        if (getHoldings(identifier).add(amount).compareTo(ess.getSettings().getMaxMoney()) <= 0) {
+            try {
+                Economy.add(identifier, amount);
+            } catch(UserDoesNotExistException | NoLoanPermittedException ignore) {
+                return false;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Used to add funds to an account.
+     * @param identifier The identifier of the account that is associated with this call.
+     * @param amount The amount you wish to add from this account.
+     * @return True if the funds were added to the account, otherwise false.
+     */
+    @Override
+    public boolean addHoldings(UUID identifier, BigDecimal amount) {
+        if (getHoldings(identifier).add(amount).compareTo(ess.getSettings().getMaxMoney()) <= 0) {
```

Comment:
I notice in a lot of these stanzas there are quite a bit of duplicated code. That should really be cut down on.